src/cast_core/include/cast_core/actors/StreamStatsReporter.hpp

```diff
@@ -0,0 +1,79 @@
+// Copyright 2023-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HEADER_D5EB1FAB_E817_43DE_A126_FCC6995B4C3C_INCLUDED
+#define HEADER_D5EB1FAB_E817_43DE_A126_FCC6995B4C3C_INCLUDED
+
+#include <string_view>
+
+#include <mongocxx/pool.hpp>
+
+#include <gennylib/Actor.hpp>
+#include <gennylib/PhaseLoop.hpp>
+#include <gennylib/context.hpp>
+
+#include <metrics/metrics.hpp>
+
+namespace genny::actor {
+
+/**
+ * Periodically polls the stats for the stream processor specified from the
+ * mstream container itself to record the metrics into genny. This is needed
+ * because stream processing is fully async so we need to rely on metrics
+ * emitted from the stream processor itself in order to get proper numbers on
+ * throughput.
+ *
+ * Throughput for a stream processor is measured based on the output rate of
+ * the stream processor.
+ *
+ * ```yaml
+ * SchemaVersion: 2017-07-01
+ * Actors:
+ * - Name: StreamStatsReporter
+ *   Type: StreamStatsReporter
+ *   Database: test
+ *   Phases:
+ *   - PollInterval: 10 seconds
```

Comment:
ik `SleepBefore` and `SleepAfter` exists, so could use that instead and just enforce that its set within the stats reporter phase config